pkg/apis/stable/v1alpha1/gameserversort.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import "sort"
+
+// GameServerSort sorts gameservers, based on a comparator
+type GameServerSort struct {
```

Comment:
overkill perhaps? sort.Slice is really fast for all practical data sizes, have you measured it and found it not to be the case?

Reply:
TL;DR - sort.Slice uses reflection. Reflection in go is _slow_. At small scale sort.Slice is probably fine, but sorting several thousand gameservers, I'll take all the performance gains we can.

Context:
https://stackoverflow.com/a/54276484

Reply:
sort.Slice uses reflection basically once or twice per call, not once per comparison (see the source code). This is only to figure out the length of the slice and generate a swapper function, after that it's not using reflection anymore.

Reply:
That's fair. I can make that change.